### PR TITLE
Fix(SCT-1103): Flaky integration tests

### PR DESCRIPTION
**Pull request by acodeninja**
### Jira Ticket

https://hackney.atlassian.net/browse/SCT-1103

### *What is the problem we're trying to solve*

When the test suite runs there is an occasional error not related to any of the code changed.

```
A total of 1 test files matched the specified pattern.
  X UpdateWorkerWithNewTeamReturnsTheOnlyTheUpdatedTeam [486ms]
  Error Message:
   Npgsql.PostgresException : 42601: syntax error at or near "Conner"
  Stack Trace:
     at Npgsql.NpgsqlConnector.<>c__DisplayClass160_0.<<DoReadMessage>g__ReadMessageLong|0>d.MoveNext()
```

[Example from CI](https://app.circleci.com/pipelines/github/LBHackney-IT/social-care-case-viewer-api/1674/workflows/5b5e22f4-76da-43ac-8814-002131142bc6/jobs/5029)

This is due to the use of `.ExecuteSqlRaw` with unescaped string interpolation. These helpers throw exceptions when `Faker` generates names like `O'Conner`.

### *What changes have we introduced*

I've removed the raw sql based seeders from the integration test helper and replaced them with calls to `Add()` and `SaveChanges()` in the entity framework.

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [ ] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ ] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### *Follow up actions after merging PR*

None

### CODE REVIEW

**Comment by Renny4Real** [comment]
@acodeninja Could we check that the test still fails when we remove/comment out the code responsible in the gateway? I encountered some false positives when we used just entity framework (similar to the above) which lead to the decision to try using the raw SQL method.

**Comment by acodeninja** [comment]
@Renny4Real I've tried commenting out the whole body of the `DatabaseGateway.UpdateWorker()` function and all tests I would expect to fail (including the integration tests) are. Is this enough?

**Comment by Renny4Real** [comment]
@acodeninja Oh nice! 
We added this test because of a bug where a worker's teams were not being updated properly and we were trying to recreate it locally. Do any of the integration tests fail if you comment out [lines 531 - 533](https://github.com/LBHackney-IT/social-care-case-viewer-api/blob/11534fa6cdad4eba449c5218596fc55660635b20/SocialCareCaseViewerApi/V1/Gateways/DatabaseGateway.cs#L531) within the gateway, as this should be the fix for the bug?

**Comment by acodeninja** [comment]
I've done some work to actually persist data to the database during integration tests, this should give us more confidence with them and avoid issues around change tracking inside a transaction for tests.

File: SocialCareCaseViewerApi.Tests/V1/IntegrationTests/IntegrationTestSetup.cs [hunk 1]
```diff
@@ -70,5 +66,14 @@ public void AfterAllTests()
         {
             _connection.Dispose();
         }
+
+        private void WipeDatabase()
+        {
+            DatabaseContext.Database.ExecuteSqlRaw("DROP SCHEMA dbo CASCADE");
+            var filePath = TestContext.CurrentContext.WorkDirectory.Replace(Path.Combine("SocialCareCaseViewerApi.Tests", "bin", "Debug", "netcoreapp3.1"), "");
```
Comment (acodeninja) [inline SocialCareCaseViewerApi.Tests/V1/IntegrationTests/IntegrationTestSetup.cs:73]: @LBHTKarki I couldn't think of a better way to load in the db schema file. Do you have any ideas?

**Review by JohnFarrellDev — APPROVED** [review]
:+1:

**Pull request merged**